src/libraries/Common/src/Internal/Cryptography/HashProvider.cs

```diff
@@ -38,6 +38,20 @@ public void AppendHashData(byte[] data, int offset, int count)
 
         public abstract bool TryFinalizeHashAndReset(Span<byte> destination, out int bytesWritten);
 
+        // Compute the hash based on the appended data and doesn't reset the hash provider to allow for continued hashing
+        public virtual int GetCurrentHash(Span<byte> destination)
+        {
+            if (destination.Length < HashSizeInBytes)
+            {
+                throw new ArgumentOutOfRangeException();
```

Comment:
All of the destination-too-small exceptions should be `ArgumentException`, not `ArgumentOutOfRangeException`, ideally with the same Argument_DestinationTooShort resource that is used elsewhere; and the paramName value specified.

https://github.com/dotnet/runtime/blob/60f2e6a21b67b6a3d6146dc4026d5adc89e284e8/src/libraries/System.Private.CoreLib/src/Resources/Strings.resx#L994-L996